netplan/cli/sriov.py

```diff
@@ -500,12 +500,12 @@ def apply_sriov_config(config_manager, rootdir='/'):
             # this only works for SR-IOV VF interfaces
             link = settings.get('link')
             vlan_id = settings.get('id')
-            if not vlan_id:
+            if not vlan_id:  # pragma: nocover (this is being reworked in PR#255)
                 raise ConfigurationError(
                     'no id property defined for SR-IOV vlan %s' % vlan)
 
             vf = vfs.get(link)
-            if not vf:
+            if not vf:  # pragma: nocover (this is being reworked in PR#255)
```

Comment:
question (non-blocking): given that #254 and #255 are pushed back for later, shouldn't we add tests for those?

This is totally your call, I just want to raise this point since the commit predates the decision.

Reply:
Yes, good point.
* For the first case (`if not vlan_id:`) we do not actually need a test, as this is now just handled a bit earlier by libnetplan, using a slightly different error message; `test_apply_sriov_config_invalid_vlan` is adopted to account for that.
* For the 2nd case (`if not vf:`) I've added a new test case.